articles/language/file-structure.md

```diff
@@ -385,3 +385,28 @@ function DotProduct(a : Double[], b : Double[]) : Double {
     return accum;
 }
 ```
+
+
+## Internal Declarations
+
+User-defined types, operations, and functions can also be declared as *internal*.
+This means that they can only be accessed from within the Q# project that they are declared in.
+When a project is used as a reference, all of its non-internal declarations are made available, but trying to use an internal declaration from another project will produce an error.
+Internal declarations are useful for writing modular code that can be reused by other parts of your project, but still be changed later without breaking other projects that might depend on it.
+
+An internal user-defined type, operation, or function can be declared simply by adding `internal` at the beginning of the declaration.
```

Comment:
It may be helpful to clarify where internal UDTs can be used; if I recall, they can't be used as a part of public signatures?

Reply:
Good idea, I added a section about that.